packages/perseus/src/renderer.tsx

```diff
@@ -550,21 +550,25 @@ class Renderer extends React.Component<Props, State> {
             // filtered out in this.render(), so we shouldn't have to
             // worry about using this widget key and ref:
             return (
-                <ErrorBoundary
+                <WidgetContainer
                     key={"container:" + id}
-                    metadata={{widget_type: type, widget_id: id}}
-                >
-                    <WidgetContainer
-                        ref={"container:" + id}
-                        type={type}
-                        initialProps={this.getWidgetProps(id)}
-                        shouldHighlight={shouldHighlight}
-                        linterContext={PerseusLinter.pushContextStack(
-                            this.props.linterContext,
-                            "widget",
-                        )}
-                    />
-                </ErrorBoundary>
+                    id={id}
+                    ref={(node) => {
+                        const containerId = "container:" + id;
```

Comment:
nitpick: Since you have to construct this ID in so many places, maybe you could make a simple function to ensure we don't get a typo somewhere?

`const containerId = (id: string) => "container:" + id;`